suite/src/benchmark.rs

```diff
@@ -0,0 +1,75 @@
+use core::arch::asm;
+
+/// Returns the machine cycle counter
+#[inline]
+pub fn get_cycle() -> u64 {
+    let counter_lo: u32;
+    let counter_hi: u32;
+
+    unsafe {
+        asm!(
+            "1: csrr {hi_old}, mcycleh",
+            "csrr {lo}, mcycle",
+            "csrr {hi}, mcycleh",
+            "bne {hi_old}, {hi}, 1b",
+            hi_old = out(reg) _,
+            lo = out(reg) counter_lo,
+            hi = out(reg) counter_hi
+        );
+    }
+    ((counter_hi as u64) << 32u64) + counter_lo as u64
+}
+
+pub mod examples {
+    use crate::platform::{self, Platform};
+
+    pub fn sha256_benchmark() {
+        if let Some(hmac_module) = platform::current().get_sha256_module() {
+            let input = [
+                0xdf3f6198, 0x04a92fdb, 0x4057192d, 0xc43dd748, 0xea778adc, 0x52bc498c, 0xe80524c0,
+                0x14b81119, 0xdf3f6198, 0x04a92fdb, 0x4057192d, 0xc43dd748, 0xea778adc, 0x52bc498c,
+                0xe80524c0, 0x14b81119, 0xdf3f6198, 0x04a92fdb, 0x4057192d, 0xc43dd748, 0xea778adc,
+                0x52bc498c, 0xe80524c0, 0x14b81119, 0xdf3f6198, 0x04a92fdb, 0x4057192d, 0xc43dd748,
+                0xea778adc, 0x52bc498c, 0xe80524c0, 0x14b81119, 0xdf3f6198, 0x04a92fdb, 0x4057192d,
+                0xc43dd748, 0xea778adc, 0x52bc498c, 0xe80524c0, 0x14b81119, 0xdf3f6198, 0x04a92fdb,
+                0x4057192d, 0xc43dd748, 0xea778adc, 0x52bc498c, 0xe80524c0, 0x14b81119, 0xdf3f6198,
+                0x04a92fdb, 0x4057192d, 0xc43dd748, 0xea778adc, 0x52bc498c, 0xe80524c0, 0x14b81119,
+                0xdf3f6198, 0x04a92fdb, 0x4057192d, 0xc43dd748, 0xea778adc, 0x52bc498c, 0xe80524c0,
+                0x14b81119,
+            ];
+            let mut output = [0u32; 8];
+
+            let cycle1 = super::get_cycle();
+            hmac_module.init_sha256();
+            let cycle2 = super::get_cycle();
+            hmac_module.write_input(&input);
+            let cycle3 = super::get_cycle();
```

Comment:
Not sure if this cycle count is necessary? It may add unnecessary overheads as the hmac computation may complete while running this? Is that possible? What do you think?